src/hooks/useSignInDialog/Steps/AccountStep.tsx

```diff
@@ -0,0 +1,148 @@
+import accountCreation from '@/assets/account-creation.png'
+import { useActiveUser, useJoystream } from '@/hooks'
+import { Placeholder, Spinner, Text } from '@/shared/components'
+import { transitions } from '@/shared/theme'
+import React, { FormEvent, useState } from 'react'
+import { CSSTransition, SwitchTransition } from 'react-transition-group'
+import {
+  AccountStepImg,
+  AccountsWrapper,
+  AccountWrapper,
+  AccountInfo,
+  IconWrapper,
+  OrderedSteps,
+  OrderedStep,
+  IconGroup,
+  AccountAddress,
+  StyledRadioButton,
+  StyledButton,
+} from './AccountStep.style'
+import polkadotIcon from '@/assets/polkadot.png'
+import joystreamIcon from '@/assets/logo.png'
+import { StepFooter, BottomBarIcon, StepSubTitle, StepTitle, StepWrapper } from './Steps.style'
+import { useNavigate } from 'react-router'
+import { SvgGlyphNewChannel, SvgOutlineConnect } from '@/shared/icons'
+
+const AccountStep: React.FC = () => {
+  const navigate = useNavigate()
+  const { setActiveUser } = useActiveUser()
+  const { accounts } = useJoystream()
+  const [imageLoaded, setImageLoaded] = useState(false)
+  const [selectedAccountAddress, setSelectedAccountAddress] = useState<undefined | string>()
+
+  const handleSubmitSelectedAccount = async (e: FormEvent) => {
+    e.preventDefault()
+    if (selectedAccountAddress) {
```

Comment:
No need to change here but the opposite generally creates less nested code and is easier to read:
```ts
if (!selectedAccountAddress) {
  return
}
...
```